scrapyrt/resources.py

```diff
@@ -147,66 +144,89 @@ def render_POST(self, request, **kwargs):
         """
         request_body = request.content.getvalue()
         try:
-            request_data = demjson.decode(request_body)
+            api_params = demjson.decode(request_body)
         except ValueError as e:
             message = "Invalid JSON in POST body. {}"
             message.format(e.pretty_description())
             raise Error('400', message=message)
 
-        log.msg("{}".format(request_data))
-        spider_data = self.get_required_argument(request_data, "request")
-        error_msg = "Missing required key 'url' in 'request' object"
-        self.get_required_argument(spider_data, "url", error_msg=error_msg)
-
-        return self.prepare_crawl(request_data, spider_data, **kwargs)
-
-    def get_required_argument(self, request_data, name, error_msg=None):
+        log.msg("{}".format(api_params))
+        try:
+            scrapy_request_args = extract_scrapy_request_args(
+                api_params.get("request", {}), raise_error=True
+            )
+        except ValueError as e:
+            raise Error(400, e.message)
+
+        if not api_params.get("start_requests"):
+            self.get_required_argument(api_params, "request")
```

Comment:
@chekunkov I cleaned this up here: https://github.com/scrapinghub/scrapyrt/pull/38/commits/59d0a0153ac041c7184da39ad139e4c0b88969eb and also added bunch of other tests. Test coverage should be much improved after this PR which should help in porting to Python3.